test/domain_test.py

```diff
@@ -21,7 +21,53 @@
 )
 
 
-class BuildJujuPodSpecTest(unittest.TestCase):
+class PrometheusServiceConfigAndPodSpecTest(unittest.TestCase):
+    def setUp(self):
+        self.default_mock_config = {
+            'external-labels': '',
+            'monitor-k8s': False,
+            'log-level': '',
+            'additional-cli-args': None,
+            'web-enable-admin-api': False,
+            'web-page-title': 'PrometheusTest',
+            'web-max-connections': 512,
+            'web-read-timeout': '5m',
+            'tsdb-retention-time': '18d',
+            'tsdb-wal-compression': True,
+            'alertmanager-notification-queue-capacity': 10000,
+            'alertmanager-timeout': '10s',
+        }
+
+    def test__cli_args_are_rendered_correctly(self):
```

Comment:
I think this can live in its own test class. The `BuildJujuPodSpecTest` should just be about the `build_juju_pod_spec`. Gotta stick to the "single responsibility" rule of classes and tests :-)

Reply:
Fixed.